src/public/view/Reset/Reset.tsx

```diff
@@ -51,12 +47,20 @@ function useResetPassword() {
   });
 }
 
+type State = 'ValidToken' | 'InvalidToken' | 'ResetSuccess' | 'ResetFailure';
+
 export function Reset() {
   const { resetToken } = resetRoute.useParams();
   const info = useTokenInfo(resetToken);
   const reset = useResetPassword();
   const loginHref = useHref(loginRoute);
 
+  const tokenState: State = info.isSuccess ? 'ValidToken' : 'InvalidToken';
```

Comment:
The whole idea of creating `State` type is such that your application remains in exactly one state. If you have two state variables, then it defeats the purpose of create single state value.